src/Model/Process.php

```diff
@@ -5,68 +5,302 @@
+		}
+
+		if (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN') {
+			$resource = proc_open('cmd /c start /b ' . $cmdline, [], $foo, self::getApp()->get_basepath());
+		} else {
+			$resource = proc_open($cmdline . ' &', [], $foo, self::getApp()->get_basepath());
+		}
+		if (!is_resource($resource)) {
+			logger('We got no resource for command ' . $cmdline, LOGGER_DEBUG);
+			return;
+		}
+		proc_close($resource);
+	}
 
-		return $return;
+	/**
+	 * Loads the process with a given ID
+	 *
+	 * @param int $id  The ID of the current process
+	 */
+	public static function load($id) {
+		if (!is_null(self::$instance)) {
+			logger('Process instance already loaded');
+			return;
+		}
+
+		$process= dba::selectFirst('process', ['pid', 'type', 'parent', 'command'], ['id' => $id]);
+
+		if (DBM::is_result($process)) {
+			self::$instance = new Process(
+				$id,
+				$process['command'],
+				$process['type'],
+				$process['pid'],
+				$process['parent']
+			);
+			self::$instance->updatePID(getmypid());
+		}
 	}
 
 	/**
-	 * Remove a process row by pid. If the pid parameter is omitted, we use the current pid
+	 * Stops the current process instance
 	 *
-	 * @param string $pid
-	 * @return bool
+	 * @param bool $kill  false = SIGTERM, true = SIGHUP
+	 * @param int  $id    The id of a process to stop
+	 *
+	 * @return bool returns true, if the process was successfully stopped
 	 */
-	public static function deleteByPid($pid = null)
+	public static function stop($kill = false, $id = null)
 	{
-		if ($pid === null) {
-			$pid = getmypid();
+		if (is_null(self::$instance) && is_null($id)) {
+			return false;
 		}
 
-		return dba::delete('process', ['pid' => $pid]);
+		if (!is_null($id)) {
+			$ret = dba::selectFirst('process', ['pid'], [ 'id' => $id ]);
+			if (DBM::is_result($ret)) {
+				posix_kill($ret['pid'], ($kill ? SIGHUP : SIGTERM));
+				return dba::delete('process', ['id' => $id]);
+			} else {
+				return false;
+			}
+		} else {
+			posix_kill(self::$instance->pid, ($kill ? SIGHUP : SIGTERM));
+			dba::delete('process', ['id' => self::$instance->id]);
+			self::$instance = null;
+			return true;
+		}
+	}
+
+	/**
+	 * Checks inactive processes based on the same parent (to avoid checking processes of other nodes)
+	 *
+	 * @param int $id The ID of the process
+	 *
+	 * @return bool Returns true if the process is inactive
+	 */
+	public static function checkInactive($id) {
+		$entry = dba::selectFirst('process', ['pid'], ['id' => $id, 'parent' => self::$instance->parent]);
+		if (DBM::is_result($entry)) {
+			return !posix_kill($entry["pid"], 0);
+		} else {
+			return false;
+		}
 	}
 
 	/**
 	 * Clean the process table of inactive physical processes
 	 */
 	public static function deleteInactive()
 	{
+		if (is_null(self::$instance) || is_null(self::$instance->parent)) {
+			return;
+		}
+
 		dba::transaction();
 
-		$processes = dba::select('process', ['pid']);
+		$processes = dba::select('process', ['id', 'pid'], ['parent' => self::$instance->parent ]);
```

Comment:
Reminder to me (Because I'm currently at work): parent could be the current instance too => add it to the selection